beacon-chain/state/state-native/validator_index_cache_test.go

```diff
@@ -0,0 +1,94 @@
+package state_native
+
+import (
+	"testing"
+
+	fieldparams "github.com/prysmaticlabs/prysm/v5/config/fieldparams"
+	"github.com/prysmaticlabs/prysm/v5/consensus-types/primitives"
+	ethpb "github.com/prysmaticlabs/prysm/v5/proto/prysm/v1alpha1"
+	"github.com/prysmaticlabs/prysm/v5/testing/require"
+)
+
+func Test_FinalizedValidatorIndexCache(t *testing.T) {
+	c := newFinalizedValidatorIndexCache()
+	b := &BeaconState{validatorIndexCache: c}
+
+	// What happens if you call getValidatorIndex with a public key that is not in the cache and state?
+	// The function will return 0 and false.
+	i, exists := b.getValidatorIndex([fieldparams.BLSPubkeyLength]byte{0})
+	require.Equal(t, primitives.ValidatorIndex(0), i)
+	require.Equal(t, false, exists)
+
+	// Validators are added to the state. They are [0, 1, 2]
+	b.validators = []*ethpb.Validator{
+		{PublicKey: []byte{1}},
+		{PublicKey: []byte{2}},
+		{PublicKey: []byte{3}},
+	}
+	// We should get able to retrieve these validators by public key even they are not in the cache
+	i, exists = b.getValidatorIndex([fieldparams.BLSPubkeyLength]byte{1})
+	require.Equal(t, primitives.ValidatorIndex(0), i)
+	require.Equal(t, true, exists)
+	i, exists = b.getValidatorIndex([fieldparams.BLSPubkeyLength]byte{2})
+	require.Equal(t, primitives.ValidatorIndex(1), i)
+	require.Equal(t, true, exists)
+	i, exists = b.getValidatorIndex([fieldparams.BLSPubkeyLength]byte{3})
+	require.Equal(t, primitives.ValidatorIndex(2), i)
+	require.Equal(t, true, exists)
+
+	// State is finalized. We flush [0, 1, 2 ] to the cache.
```

Comment:
```suggestion
	// State is finalized. We save [0, 1, 2] to the cache.
```
This is nitpicky, but technically "flush" would mean these indices are no longer present in the state. Someone reading the test might get the wrong impression of what the code does.